test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -288,6 +298,40 @@ test_ucp_ep_reconfig::entity::get_address(const ucp_tl_bitmap_t &tl_bitmap) cons
     return address;
 }
 
+void test_ucp_ep_reconfig::init_buffers(mem_buffer_vec_t &sbufs,
+                                        mem_buffer_vec_t &rbufs,
+                                        mem_buffer_vec_t &o_sbufs,
+                                        mem_buffer_vec_t &o_rbufs, size_t size,
+                                        bool bidirectional)
+{
+    for (unsigned i = 0; i < num_iterations; ++i) {
+        sbufs.push_back(
+                mem_buffer_p(new mem_buffer(size, UCS_MEMORY_TYPE_HOST, i)));
+        rbufs.push_back(mem_buffer_p(
+                new mem_buffer(size, UCS_MEMORY_TYPE_HOST, ucs::rand())));
+
+        if (bidirectional) {
+            o_sbufs.push_back(mem_buffer_p(
+                    new mem_buffer(size, UCS_MEMORY_TYPE_HOST, i)));
+            o_rbufs.push_back(mem_buffer_p(
+                    new mem_buffer(size, UCS_MEMORY_TYPE_HOST, ucs::rand())));
+        }
+    }
+}
+
+void test_ucp_ep_reconfig::pattern_check(const mem_buffer_vec_t &rbufs,
+                                         const mem_buffer_vec_t &o_rbufs,
+                                         size_t size, bool bidirectional) const

```

Comment:
do we really have to pass the size? pattern_check used to be called w/o size param, guess size was taken from the buffer

Reply:
the buffers are initialized with max_buffer_size:
```
        init_buffers(sbufs, rbufs, o_sbufs, o_rbufs,
                     msg_sizes[ucs_static_array_size(msg_sizes)-1],
                     bidirectional);
```
so need to pass real size from outside

Reply:
we could move buffers declaration and initialization inside the `for (auto msg_size : msg_sizes) {` loop, as it used to be, then the size would be exact and not need to take the last size in the array

Reply:
ok, but notice it colides with https://github.com/openucx/ucx/pull/10331#discussion_r1865682440